openbb_platform/providers/coingecko/openbb_coingecko/models/crypto_price.py

```diff
@@ -0,0 +1,169 @@
+    CryptoPriceQueryParams,
+)
+from openbb_core.provider.utils.descriptions import QUERY_DESCRIPTIONS
+from openbb_core.provider.utils.errors import EmptyDataError
+from openbb_coingecko.utils.helpers import make_request, validate_symbol
+from pydantic import Field, field_validator
+
+
+class CoinGeckoCryptoPriceQueryParams(CryptoPriceQueryParams):
+    """CoinGecko Real-time Crypto Price Query.
+
+    Source: https://docs.coingecko.com/reference/simple-price
+    """
+
+    vs_currency: str = Field(
+        default="usd",
+        description="The target currency of market data (usd, eur, jpy, etc.)",
+    )
+    include_market_cap: bool = Field(
+        default=True,
+        description="Include market cap in the response.",
+    )
+    include_24hr_vol: bool = Field(
+        default=True,
+        description="Include 24hr volume in the response.",
+    )
+    include_24hr_change: bool = Field(
+        default=True,
+        description="Include 24hr change in the response.",
+    )
+    include_last_updated_at: bool = Field(
+        default=True,
+        description="Include last updated timestamp in the response.",
+    )
+    precision: Optional[str] = Field(
+        default=None,
+        description=(
+            "The precision of the data. Use 'full' for full precision, "
+            "otherwise 2 decimals."
+        ),
+    )
+
+    @field_validator("vs_currency", mode="before", check_fields=False)
+    @classmethod
+    def validate_vs_currency(cls, v: str) -> str:
+        """Validate and normalize vs_currency."""
+        return v.lower().strip()
+
+    @field_validator("symbol", mode="before", check_fields=False)
+    @classmethod
+    def validate_symbol(cls, v: Union[str, List[str]]) -> str:
+        """Validate and normalize symbol(s)."""
+        if isinstance(v, str):
+            return validate_symbol(v)
+        return ",".join([validate_symbol(symbol) for symbol in v])
+
+
+class CoinGeckoCryptoPriceData(CryptoPriceData):
+    """CoinGecko Real-time Crypto Price Data."""
+
+
+class CoinGeckoCryptoPriceFetcher(
+    Fetcher[
+        CoinGeckoCryptoPriceQueryParams,
+        List[CoinGeckoCryptoPriceData],
+    ]
+):
+    """Transform the query, extract and transform the data from the CoinGecko endpoints."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> CoinGeckoCryptoPriceQueryParams:
+        """Transform the query parameters."""
+        return CoinGeckoCryptoPriceQueryParams(**params)
+
+    @staticmethod
+    async def aextract_data(
+        query: CoinGeckoCryptoPriceQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Return the raw data from the CoinGecko endpoint."""
+        api_key = credentials.get("coingecko_api_key") if credentials else None
+        
+        symbols = query.symbol.split(",")
+        
+        # Build parameters for the API call
+        params = {
+            "ids": ",".join(symbols),
```

Comment:
No need to make this a list to only convert back to a string. Use the attribute directly.